src/lib/Scenes/Show/Components/Artists/Artists.tsx

```diff
@@ -0,0 +1,130 @@
+import { Sans, Spacer } from "@artsy/palette"
+import { InvertedButton } from "lib/Components/Buttons"
+import { get, take } from "lodash"
+import React from "react"
+import { commitMutation, createFragmentContainer, graphql, RelayProp } from "react-relay"
+import { ArtistListItem } from "./Components/ArtistListItem"
+
+import { Artists_show } from "__generated__/Artists_show.graphql"
+import { ArtistsFollowArtistMutation } from "__generated__/ArtistsFollowArtistMutation.graphql"
+
+interface Props {
+  show: Artists_show
+  relay: RelayProp
+}
+
+interface State {
+  isExpanded: boolean
+  isFollowedChanging: { [id: string]: boolean }
+}
+
+export class Artists extends React.Component<Props, State> {
+  state = {
+    isExpanded: false,
+    isFollowedChanging: {},
+  }
+
+  handleFollowArtist = ({ id, __id, is_followed }) => {
+    const { relay } = this.props
+    const { isFollowedChanging } = this.state
+    this.setState(
+      {
+        isFollowedChanging: {
+          ...isFollowedChanging,
+          [id]: true,
+        },
+      },
+      () => {
+        commitMutation<ArtistsFollowArtistMutation>(relay.environment, {
+          onCompleted: () => {
+            this.setState({
+              isFollowedChanging: {
+                ...this.state.isFollowedChanging,
+                [id]: false,
+              },
+            })
+          },
+          mutation: graphql`
+            mutation ArtistsFollowArtistMutation($input: FollowArtistInput!) {
+              followArtist(input: $input) {
+                artist {
+                  __id
+                  is_followed
+                }
+              }
+            }
+          `,
+          variables: {
+            input: {
+              artist_id: id,
+              unfollow: is_followed,
+            },
+          },
+          optimisticResponse: {
+            followArtist: {
+              artist: {
+                __id,
+                is_followed: !is_followed,
+              },
+            },
+          },
+          updater: store => {
```

Comment:
Not sure if I need  both`updater` and `optimisticResponse`, I saw both used in some places and in others only one...

Reply:
I believe the optimisticResponse should be enough but maybe there's a good reason for using both